tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -447,8 +452,10 @@ static uint32_t write_pages_to_dispatcher(
 
     // Grabbing all pages at once is ok if scratch_size < 3 * downstream_cb_block_size
     // test_for_nonzero is an optimization: inner loops moving lots of pages don't bother
-    if (!test_for_nonzero || npages != 0) {
-        cb_acquire_pages<my_noc_xy, my_downstream_cb_sem_id>(npages);
+    if constexpr (!test_for_nonzero) {
+        if (npages != 0) {
+            cb_acquire_pages<my_noc_xy, my_downstream_cb_sem_id>(npages);
```

Comment:
This doesn't have parity with the original changes

Reply:
Thanks for catching this. Missed the ||